inspirehep/records/marshmallow/literature/common/external_system_identifier.py

```diff
@@ -0,0 +1,106 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2019 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+
+
+from marshmallow import Schema, fields, missing, post_dump
+
+
+class ExternalSystemIdentifierSchemaV1(Schema):
+    url_name = fields.Method("get_url_name")
+    url_link = fields.Method("get_url_link")
+
+    schema_to_url_name_map = {
+        "ads": "ADS Abstract Service",
+        "cds": "CERN Document Server",
+        "euclid": "Project Euclid",
+        "hal": "HAL Archives Ouvertes",
+        "kekscan": "KEK scanned document",
+        "msnet": "AMS MathSciNet",
+        "osti": "OSTI Information Bridge Server",
+        "zblatt": "zbMATH",
+    }
+
+    schema_to_url_link_prefix_map = {
+        "ads": "http://adsabs.harvard.edu/abs/",
+        "cds": "http://cds.cern.ch/record/",
+        "euclid": "http://projecteuclid.org/",
+        "hal": "https://hal.archives-ouvertes.fr/",
+        "kekscan": "https://lib-extopc.kek.jp/preprints/PDF/",
+        "msnet": "http://www.ams.org/mathscinet-getitem?mr=",
+        "osti": "https://www.osti.gov/scitech/biblio/",
+        "zblatt": "http://www.zentralblatt-math.org/zmath/en/search/?an=",
+    }
+
+    def get_url_name(self, data):
+        schema = data.get("schema").lower()
+        url_name = self.schema_to_url_name_map.get(schema)
+        if url_name is None:
+            return missing
+        return url_name
+
+    def get_url_link(self, data):
+        schema = data.get("schema").lower()
+        value = data.get("value")
+        url_prefix = self.schema_to_url_link_prefix_map.get(schema)
+        if url_prefix is None:
+            return missing
+        if schema == "kekscan":
+            kekscan_link = self.get_link_for_kekscan_schema(value)
+            if kekscan_link is None:
+                return missing
+            return kekscan_link
+        return "{}{}".format(url_prefix, value)
+
+    def get_link_for_kekscan_schema(self, external_system_id_value):
+        extid = external_system_id_value.replace("-", "")
+        if (
+            len(extid) == 7
```

Comment:
This is another place where would be nice to have some docstrings...Where this 7 come from, for example?